### PR TITLE
Fixes 253: removing multiple commit on documentation generation using CI

### PR DESCRIPTION
**Pull request by sch00lb0y**
## Description

Right now documentation is pushed for each webhook request. so when we push to gh-pages also we getting webhook request then ci making one more deploy and it keeps on going so I changed the `webhook` handler to do the generation if any changes in the master branch
## Types of changes

- [x] Bug fix 
- [ ] New feature

## Checklist:

- [x] I have read the [Contribution & Best Practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by sch00lb0y** [comment]
deployment https://demo08.herokuapp.com

File: routes/ci.js [hunk 1]
```diff
@@ -78,47 +78,55 @@ router.post('/register', function (req, res, next) {
...
-            deploy.deployPages({}, {
-              email: result.email,
-              gitURL: req.body.repository.clone_url,
-              username: result.username,
-              uniqueId: generatedData.uniqueId,
-              encryptedToken: result.accessToken
-            })
+  var event = req.get('X-GitHub-Event');
+  var branch = req.body.ref.split("/")[2];
+  if (branch === "master") {
```
Comment (imujjwal96) [inline routes/ci.js]: should you not have use 
`if (branch !== "gh-pages")`

Comment (pri22296) [inline routes/ci.js]: Wouldn't that mean that all branches are built by default? should that be the default behaviour?

Comment (imujjwal96) [inline routes/ci.js]: Are you saying that the documentation will be built on all other branches irrelevant of their involvement with the documentation generation process?

Comment (pri22296) [inline routes/ci.js]: Could you specify what are planning to add to the UI related to adding branches? That would make a lot of this clear.

Comment (imujjwal96) [inline routes/ci.js]: In the initial form we will add an optional field for the branch name. If the user does not specify the branch name, then their default branch should be used

Comment (pri22296) [inline routes/ci.js]: That does mean that if the default branch is let's say `development`, and the user doesn't specify anything, then builds should not occur on pushes to `master` right? But then using `if (branch !== gh-pages)`, builds would be pushed even if commit to master is made. Am i missing something?

Comment (imujjwal96) [inline routes/ci.js]: No you are not! I want to maintain the flow of development. I didn't request a change that involves branch specific build. This change will cover the ground unless https://github.com/fossasia/yaydoc/issues/251 is resolved.

Comment (pri22296) [inline routes/ci.js]: cool

Comment (sch00lb0y) [inline routes/ci.js]: made changes :smile:

**Pull request merged**